src/main/java/com/guciowons/shoppingbasket/Product/Product.java

```diff
@@ -1,10 +1,15 @@
 package com.guciowons.shoppingbasket.Product;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
 import java.math.BigDecimal;
 import java.util.HashMap;
 import java.util.Map;
 
+@Document
 public class Product {
 
+    @Id
     private Integer id;
```

Comment:
Tutaj teraz mieszasz dwa IDki, jeden Twój bazodanowy i jeden z zewnętrznego API. Co jak będziesz potem pobierał produkty z dwóch providerów i tam będą konflikty na IDkach? Co jak provider zmieni ID produktu? To powinny być dwa osobne pola - id i externalId czy coś takiego, żeby mieć nad tym większą kontrolę.